app/services/UIBuilder.scala

```diff
@@ -75,8 +75,8 @@ class UIBuilder {
       case (c: Callout) :: xs => fromStanzas(xs, acc ++ fromCallout(c, errStrategy), errStrategy)
       case (in: Input) :: _ => fromStanzas(Nil, Seq(fromInput(in, acc)), errStrategy)
       case (q: Question) :: _ => fromStanzas(Nil, Seq(fromQuestion(q, acc)), errStrategy)
-      case (ns: NonExclusiveSequence) :: _ => fromStanzas(Nil, Seq(fromNonExclusiveSequence(ns, acc)), errStrategy)
-      case (es: ExclusiveSequence) :: _ => fromStanzas(Nil, Seq(fromExclusiveSequence(es, acc)), errStrategy)
+      case (ns: NonExclusiveSequence) :: _ => fromStanzas(Nil, Seq(fromSequence(ns, acc)), errStrategy)
```

Comment:
as discussed, given you've simplified to a single core Sequence class, could do someting similar with the teo UI classes